directory/src/main/java/org/entcore/directory/controllers/UserController.java

```diff
@@ -781,14 +777,39 @@ public void ignoreDuplicate(final HttpServerRequest request) {
 		userService.ignoreDuplicate(userId1, userId2, defaultResponseHandler(request));
 	}
 
+	/**
+	 * Merges two users.
+	 *
+	 * @deprecated
+	 *
+	 * Calls to this method should be replaced by calls to {@link UserController#mergeDuplicateV2(HttpServerRequest)}
+	 * as it allows callers to specify whether or not relationships should be kept
+	 *
+	 * @param request
+	 */
+	@Deprecated
 	@Put("/duplicate/merge/:userId1/:userId2")
 	@ResourceFilter(AdmlOfTwoUsers.class)
 	@SecuredAction(value = "", type = ActionType.RESOURCE)
 	@MfaProtected()
 	public void mergeDuplicate(final HttpServerRequest request) {
 		final String userId1 = request.params().get("userId1");
 		final String userId2 = request.params().get("userId2");
-		userService.mergeDuplicate(userId1, userId2, defaultResponseHandler(request));
+		userService.mergeDuplicate(userId1, userId2, false, defaultResponseHandler(request));
+	}
+
+	@Put("/duplicate/merge")
+	@ResourceFilter(AdmlOfTwoUsers.class)
```

Comment:
Du coup, ce filtre ne doit plus fonctionner correctement, car les identifiants de users ne sont plus présents dans la route.

`public class AdmlOfTwoUsers extends AdmlResourcesProvider {`
`	@Override`
`	public void authorizeAdml(HttpServerRequest resourceRequest, Binding binding,`
`			UserInfos user, UserInfos.Function adminLocal, Handler<Boolean> handler) {`
`		String userId1 = resourceRequest.params().get("userId1");`
`		String userId2 = resourceRequest.params().get("userId2");`
`		String query =`
`				"MATCH (u:User {id: {userId}})-[:IN]->(:Group)-[:DEPENDS]->(s:Structure) " +`
`				"WHERE s.id IN {structures} " +`
`				"RETURN count(*) > 0 as exists ";`